File: pe/nonvisualstudio/elixir/Spec/Problem0029.Spec.ex
# IN POWERSHELL
# iex.bat - REPL
# elixir c:\Users\Thomas\Documents\GitHub\practice\pe\nonvisualstudio\elixir\Spec\Problem0029.Spec.ex

Code.require_file "../Implementation/Problem0029.ex", __DIR__

ExUnit.start

defmodule Problem0029Tests do
  use ExUnit.Case, async: true

  test "exponentiateNumberFromTwoToX 2 5 should return a Set of [4,8,16,32]." do
    expected = [4,8,16,32]

    assert HashSet.equal?(Problem0029.exponentiateNumberFromTwoToX(2,5), createSet(expected))
  end

  test "exponentiateNumberFromTwoToX 3 5 should return a Set of [9, 27, 81, 243]." do
    expected = [9, 27, 81, 243]

    assert HashSet.equal?(Problem0029.exponentiateNumberFromTwoToX(3,5), createSet(expected))
  end

  test "exponentiateNumberFromTwoToX 5 5 should return a Set of [25, 125, 625, 3125]." do
    expected = [25, 125, 625, 3125]

    assert HashSet.equal?(Problem0029.exponentiateNumberFromTwoToX(5,5), createSet(expected))
  end

  defp createSet(list), do: (Enum.into list, HashSet.new)

  test "distinctPowersBetween2And 5 should return 15." do
    assert Problem0029.distinctPowersBetween2And(5) === 15
  end

  test "distinctPowersBetween2And 4 should return 8." do
    assert Problem0029.distinctPowersBetween2And(4) === 8
  end

  test "distinctPowersBetween2And 100 should return 9183." do
    assert Problem0029.distinctPowersBetween2And(100) === 9183
  end
end

File: pe/nonvisualstudio/clojure/Spec/Problem0029.Spec.clj
; In powershell c:\Users\Thomas\Documents\GitHub\practice\clojureREPLRunner.bat
;(load-file "C:/GitHub/practice/pe/nonvisualstudio/clojure/Spec/Problem0029.Spec.clj")

(ns pe.clojure.implementation.problem0029.spec)

(load-file "C:/GitHub/practice/pe/nonvisualstudio/clojure/Implementation/Problem0029.clj")
(use 'pe.clojure.implementation.problem0029)
(use 'clojure.test)


(deftest Problem0029
  (testing "exponentiateNumberFromTwoToX"
    (is (= #{4 8 16 32} (exponentiateNumberFromTwoToX 2 5)))
    (is (= #{9 27 81 243} (exponentiateNumberFromTwoToX 3 5)))
    (is (= #{25 125 625 3125} (exponentiateNumberFromTwoToX 5 5))))
  (testing "distinctPowersBetween2And"
    (is (= 15 (distinctPowersBetween2And 5)))
    (is (= 8 (distinctPowersBetween2And 4)))
    (is (= 9183 (distinctPowersBetween2And 100)))
    ))


(run-tests)
